### PR TITLE
Add support for `ephemeral` storageClass

### PR DESCRIPTION
**Pull request by amisevsk**
### What does this PR do?
Adds support for the `ephemeral` storageClass (use emptyDir for all DevWorkspace volumes). This is useful (at least) for testing non-storage related usecases of DWO, as it avoids the need for a cleanup pod on every DevWorkspace. 

Currently, the implementation is very simple and does no validation (e.g. that all container components reference existing volumes). 

### What issues does this PR fix or reference?
No issue created AFAIK but was very useful while testing https://github.com/devfile/devworkspace-operator/pull/492

### Is it tested? How?
Set `.spec.template.attributes.controller.devfile.io/storage-type: "ephemeral"` and start workspace.

### PR Checklist

- [ ] E2E tests pass (when PR is ready, comment `/test v7-devworkspaces-operator-e2e, v7-devworkspace-happy-path` to trigger)
    - [ ] `v7-devworkspaces-operator-e2e`: DevWorkspace e2e test
    - [ ] `v7-devworkspace-happy-path`: DevWorkspace e2e test

### CODE REVIEW

File: pkg/provision/storage/ephemeralStorage.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+import (
+	dw "github.com/devfile/api/v2/pkg/apis/workspaces/v1alpha2"
+
+	"github.com/devfile/devworkspace-operator/apis/controller/v1alpha1"
+	"github.com/devfile/devworkspace-operator/controllers/workspace/provision"
+	"github.com/devfile/devworkspace-operator/pkg/library/container"
+)
+
+// The CommonStorageProvisioner provisions one PVC per namespace and configures all volumes in a workspace
```
Comment (sleshchenko) [inline pkg/provision/storage/ephemeralStorage.go:23]: Please update this

**Review by sleshchenko — APPROVED** [review]

File: pkg/provision/storage/ephemeralStorage.go (hunk 1 above)
Comment (amisevsk) [reply pkg/provision/storage/ephemeralStorage.go:23]: Woops :D 

My copy-paste is discovered

**Review by JPinkney — APPROVED** [review]

**Pull request merged**